release-manager/files/README.md

```diff
@@ -128,8 +179,180 @@ services:
+Similarly, the number of retries is configurable with the `openshiftRolloutTimeoutRetries` property.
+
+The following example establishes a timeout of `120` minutes for both `qa` and `prod` environments with a total number of `3` retries.
+
+```yaml
+...
+environments:
+  prod:
+    apiUrl: https://...
+    credentialsId: ...
+    openshiftRolloutTimeoutMinutes: 120
+    openshiftRolloutTimeoutRetries: 3
+  qa:
+    openshiftRolloutTimeoutMinutes: 120
+    openshiftRolloutTimeoutRetries: 3
+...
+```
+
+### Walkthrough
+
+Let's start by assuming you have a project FOO with two components, X and Y. These components are defined under the `repositories` section in the `metadata.yml` file of the release manager repository. When you want to create a new release, you start the orchestration pipeline with input parameters - we will use version `1` and change ID `1234` in this example. The environment should be `DEV`. At the end of the pipeline run, you'll have a release bundle identified by the tag `v1-1234-0-D`. This release can later be promoted as-is to QA. Once it is installed there, the same release bundle will be tagged with `v1-1234-0-Q` which can then be promoted to PROD (where it will be tagged with `v1-1234-0-P`).
+
+To create a release bundle, the orchestration pipeline will first trigger the build of each component. Then, it will export all resources in your OpenShift namespace (`$PROJECT-$ENVIRONMENT`, here `foo-dev`) belonging to the component. By convention, this means all resources labeled with `app=$PROJECT-$COMPONENT` (e.g. `app=foo-x`). Any resources without such a label will NOT be part of the release bundle. The exported resources are stored in a `template.yml` file (an OpenShift template) located in the `openshift-exported` folder within each component repository. Further, the container image SHA of the running pod is retrieved and stored in the file `image-sha` in the same folder. Once done, the orchestration pipeline will commit the two files, tag the commit with `v1-1234-0-D` and push to the remote. After this process has been done for all repositories, the same tag is also applied to the release manager repository. At this stage, the "dev release bundle" is complete and can be installed into QA.
+
+To trigger the installation of an existing release bundle, the user needs to supply a `version` and `changeId` which has previously been used to create a release bundle. In our example, supplying `version=1`, `changeId=1234` and `environment=QA` will promote the release bundle identified by `v1-1234-0-D` to the QA environment and tag it with `v1-1234-0-Q`. Now that we have a "QA release bundle", we can promote it to PROD by supplying `version=1`, `changeId=1234` and `environment=PROD`.
+
+
+### Customizing release bundle creation
+
+As outlined above, a release bundle is essentially a state of all involved Git repositories. Each component repository contains two artifacts:
+
+- a container image SHA
```

Comment:
this only applies to specific types ods*, no?
Does it apply to ta helm deployment?

Reply:
It also applied to helm components. The release manager will "keep" the image sha in a file that will be used later in order to move it accross environments. It is one of the reasons we force deployments and DeploymentConfigs for ods and ods-service type